libV2/tasks/BaseTask.js

```diff
@@ -39,6 +46,70 @@ class BaseTask extends Process {
                 .option('-l, --lag <lag>', 'Set a custom lag time in seconds', v => parseInt(v, 10))
                 .option('-n, --node-id <id>', 'Set a custom node id');
         }
+
+        if (this._enableMetrics) {
+            this._metricsHandlers = {
+                ...this._registerDefaultMetricHandlers(),
+                ...this._registerMetricHandlers(),
+            };
+            await this._createProbeServer();
+        }
+    }
+
+    _registerDefaultMetricHandlers() {
+        const taskName = this.constructor.name;
+
+        // Get the name of our subclass in snake case format eg BaseClass => _base_class
+        const taskNameSnake = taskName.replace(/[A-Z]/g, letter => `_${letter.toLowerCase()}`);
+
+        const executionTime = new promClient.Gauge({
+            name: `utapi${taskNameSnake}_execution_seconds`,
```

Comment:
* `execution_seconds` seems a bit strange, maybe `duration_seconds` would be more self-describing?
* instead of a gauge, it may be better to use two counter: `total_duration` and `count`, which can be much more easily managed afterwards (aggregated, mean...); to simplify this, an histogram or summary may be used.